cmd/soci/commands/rebuild_db.go

```diff
@@ -19,16 +19,15 @@ package commands
 import (
 	"path/filepath"
 
-	"github.com/awslabs/soci-snapshotter/config"
 	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/awslabs/soci-snapshotter/soci/store"
 	"github.com/containerd/containerd/cmd/ctr/commands"
 	"github.com/urfave/cli"
-	"oras.land/oras-go/v2/content/oci"
 )
 
 var RebuildDBCommand = cli.Command{
 	Name:  "rebuild-db",
-	Usage: `rebuild the artifacts database. You should use this command after "rpull" so that indices/ztocs can be discovered by commands like "soci index list".`,
+	Usage: `rebuild the artifacts database. You should use this command after "rpull" so that indices/ztocs can be discovered by commands like "soci index list", and after "index rm" when using the containerd content store so that deleted orphaned zTOCs will be forgotten`,
```

Comment:
Just want to clarify - If we `soci index rm` an index than it's children will be _eligible_ for deletion. Is there a scenario where we delete an index and then rebuild the db but the ztocs still persists since the GC hasn't run yet?

Reply:
Yes. Garbage collection frequency and triggers are controlled by containerd. Here are some scenarios where garbage collection wouldn't be triggered:
1. The user has specified custom gc configuration for their containerd that adds delays, lowers frequency, etc
2. Containerd started less than 100ms ago, that's the default startup delay before the first gc
3. Garbage collection ran recently; default containerd configuration allows gc to take 2% of total runtime so if gc takes 1ms then there's a 49ms gap where no gc will take place.

Reply:
There are mechanisms to perform synchronous delete and garbage collection for images and leases, but not for content. https://github.com/containerd/containerd/blob/main/docs/garbage-collection.md#synchronous-garbage-collection
If that ever gets extended to include content, we could take advantage of it.

Reply:
Cool thanks for the clarification